starky/src/verifier.rs

```diff
@@ -88,8 +90,18 @@ where
         l_1,
         l_last,
     );
-    stark.eval_ext(vars, &mut consumer);
-    // TODO: Add in constraints for permutation arguments.
+    let permutation_data = stark.uses_permutation_args().then(|| PermutationCheckData {
+        local_zs: permutation_zs.as_ref().unwrap().clone(),
```

Comment:
I think we should return an error if the STARK uses permutation args but those openings are missing (or the challenges are missing because the cap was missing), vs panicking.

Reply:
What do you think about adding a function
```rust
fn check_permutation_options<
    F: RichField + Extendable<D>,
    C: GenericConfig<D, F = F>,
    S: Stark<F, D>,
    const D: usize,
>(
    stark: &S,
    proof_with_pis: &StarkProofWithPublicInputs<F, C, D>,
    challenges: &StarkProofChallenges<F, D>,
) -> Result<()> {
    let options = [
        proof_with_pis.proof.permutation_zs_cap.is_some(),
        proof_with_pis.proof.openings.permutation_zs.is_some(),
        proof_with_pis.proof.openings.permutation_zs_right.is_some(),
        challenges.permutation_challenge_sets.is_some(),
    ];
    ensure!(
        options
            .into_iter()
            .all(|b| b == stark.uses_permutation_args()),
        "Permutation data doesn't match with Stark configuration."
    );
    Ok(())
}
```
at the top of the verifier?

Reply:
Sounds good, good idea 👍